app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,45 @@ def index
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def create_comment
+    @comment = Comment.new(comment_prarams)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:id])
+    @comment.save
+    redirect_to user_post_path
+  end
+
+  def create_like
+    @like = Like.new(author: current_user, post: Post.find(params[:id]))
+    @like.save
+    redirect_to user_post_path
+  end
+
```

Comment:
- [ ] The code for creating comments and likes should be in the respective comments and likes controllers. Kindly use the `new` and `create` in the comments and likes controllers.

- [ ] Kindly remove these lines on 9 and 10:

  ```
  @comment = Comment.new
  @like = Like.new
  ```
  Move them to the `new` actions in their respective controllers.

Reply:
Well resolved ✔️